userspace/libpman/src/maps.c

```diff
@@ -75,76 +75,159 @@ uint64_t pman_get_probe_schema_ver() {
 
 /*=============================== BPF GLOBAL VARIABLES ===============================*/
 
+int pman_get_capture_settings(struct capture_settings* settings) {
+	char error_message[MAX_ERROR_MESSAGE_LEN];
+	int ret;
+	uint32_t key = 0;
+	int fd = bpf_map__fd(g_state.skel->maps.capture_settings);
+	if(fd <= 0) {
+		snprintf(error_message, MAX_ERROR_MESSAGE_LEN, "unable to get capture_settings map fd!");
+		pman_print_error((const char*)error_message);
+		return errno;
+	}
+	if((ret = bpf_map_lookup_elem(fd, &key, settings)) != 0) {
+		snprintf(error_message, MAX_ERROR_MESSAGE_LEN, "unable to get capture_settings!");
+		pman_print_error((const char*)error_message);
+	}
+
+	return ret;
+}
+
+int pman_update_capture_settings(struct capture_settings* settings) {
+	char error_message[MAX_ERROR_MESSAGE_LEN];
+	int ret;
+	int fd = bpf_map__fd(g_state.skel->maps.capture_settings);
+	if(fd <= 0) {
+		snprintf(error_message, MAX_ERROR_MESSAGE_LEN, "unable to get capture_settings map fd!");
+		pman_print_error((const char*)error_message);
+		return errno;
+	}
+	uint32_t key = 0;
+	if((ret = bpf_map_update_elem(fd, &key, settings, BPF_ANY)) != 0) {
+		snprintf(error_message,
+		         MAX_ERROR_MESSAGE_LEN,
+		         "unable to initialize capture_settings map!");
+		pman_print_error((const char*)error_message);
+	}
+
+	return ret;
+}
+
 void pman_set_snaplen(uint32_t desired_snaplen) {
-	g_state.skel->bss->g_settings.snaplen = desired_snaplen;
+	struct capture_settings settings;
+	pman_get_capture_settings(&settings);
```

Comment:
I would return here if we are not able to get the settings. There is the risk we will write junk at the next `update`
```suggestion
	struct capture_settings settings = { 0 };
	if(pman_get_capture_settings(&settings)!=0){return;}
```

Reply:
Makes sense, done!